pkg/cmd/repo/view/view.go

```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return err
+	}
+
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	repo, err := api.GitHubRepo(apiClient, toView)
+	if err != nil {
+		return err
+	}
+
+	openURL := generateRepoURL(toView, "")
+	if opts.Web {
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", displayURL(openURL))
+		}
+		return utils.OpenInBrowser(openURL)
+	}
+
+	fullName := ghrepo.FullName(toView)
+
+	readme, err := RepositoryReadme(apiClient, toView)
+	var notFound *api.NotFoundError
+	if err != nil && !errors.As(err, &notFound) {
+		return err
+	}
+
+	stdout := opts.IO.Out
+
+	if !opts.IO.IsStdoutTTY() {
+		fmt.Fprintf(stdout, "name:\t%s\n", fullName)
+		fmt.Fprintf(stdout, "description:\t%s\n", repo.Description)
+		if readme != nil {
+			fmt.Fprintln(stdout, "--")
+			fmt.Fprintf(stdout, readme.Content)
+			fmt.Fprintln(stdout)
+		}
+
+		return nil
+	}
+
+	repoTmpl := heredoc.Doc(`
+		{{.FullName}}
+		{{.Description}}
+		
+		{{.Readme}}
+		
+		{{.View}}
+		`)
+
+	tmpl, err := template.New("repo").Parse(repoTmpl)
+	if err != nil {
+		return err
+	}
+
+	var readmeContent string
+	if readme == nil {
+		readmeContent = utils.Gray("This repository does not have a README")
+	} else if isMarkdownFile(readme.Filename) {
+		var err error
+		readmeContent, err = utils.RenderMarkdown(readme.Content)
+		if err != nil {
+			return fmt.Errorf("error rendering markdown: %w", err)
+		}
+	} else {
+		readmeContent = readme.Content
+	}
+
+	description := repo.Description
+	if description == "" {
+		description = utils.Gray("No description provided")
+	}
+
+	repoData := struct {
+		FullName    string
+		Description string
+		Readme      string
+		View        string
+	}{
+		FullName:    utils.Bold(fullName),
+		Description: description,
+		Readme:      readmeContent,
+		View:        utils.Gray(fmt.Sprintf("View this repository on GitHub: %s", openURL)),
+	}
+
+	err = tmpl.Execute(stdout, repoData)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isMarkdownFile(filename string) bool {
+	// kind of gross, but i'm assuming that 90% of the time the suffix will just be .md. it didn't
+	// seem worth executing a regex for this given that assumption.
+	return strings.HasSuffix(filename, ".md") ||
+		strings.HasSuffix(filename, ".markdown") ||
+		strings.HasSuffix(filename, ".mdown") ||
+		strings.HasSuffix(filename, ".mkdown")
+}
+
+// TODO COPYPASTA FROM command; CONSIDER FOR cmdutil?
+func isURL(arg string) bool {
+	return strings.HasPrefix(arg, "http:/") || strings.HasPrefix(arg, "https:/")
+}
+
+func generateRepoURL(repo ghrepo.Interface, p string, args ...interface{}) string {
+	baseURL := fmt.Sprintf("https://%s/%s/%s", repo.RepoHost(), repo.RepoOwner(), repo.RepoName())
+	if p != "" {
+		return baseURL + "/" + fmt.Sprintf(p, args...)
+	}
+	return baseURL
+}
+
+func displayURL(urlStr string) string {
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return urlStr
+	}
+	return u.Hostname() + u.Path
+}
```

Comment:
Perhaps these can be extracted somewhere in a separate package and imported to any command that needs them?

I don't think that `cmdutil` is the right package for them since "cmdutil" wants to hold helpers specific to shared _command_ behavior. Since these helpers do not pertain to Cobra commands specifically, maybe we can invent a more suitable place for them. For instance, `generateRepoURL` feels like it might want to belong to `ghrepo`.

Reply:
RepoURL felt fine in ghrepo. I was resistant to putting more in `utils` but it seemed like the best place for now for the other two.